mirrord-cli/src/main.rs

```diff
@@ -138,7 +141,7 @@ fn main() {
     match cli.commands {
         Commands::Exec(args) => exec(&args).unwrap(),
         Commands::Extract { path } => {
-            extract_library(Some(path));
+            extract_library(Some(path)).unwrap();
```

Comment:
Change fn main to return `Result<()>` then you can use `?` here

Reply:
If I change the main function signature to return `Result<()>`, then we'd have to remove the `unwrap()`s/`?` to return the `Result` - or we can leave in the `unwrap()`s/`?` and leave out `Result` from the return type - which behavior would be preferred?

Reply:
`?` is a syntatic sugar that returns when a `Result`/`Option` has `Err`/`None` else it returns the inner value.
So the change would to add `?` and change the return type to `Result`
https://doc.rust-lang.org/rust-by-example/std/result/question_mark.html

Reply:
I understand that, but the issue is that, if we leave in the `Result` return type, running cargo check gives me the following:

<a href="https://ibb.co/xFV8zKR"><img src="https://i.ibb.co/Sfq7VD9/Screen-Shot-2022-06-03-at-2-27-25-PM.png" alt="Screen-Shot-2022-06-03-at-2-27-25-PM" border="0"></a>

So the types don't match up if we add `Result` alongside unwrapping. I just want to make sure I'm making the changes you want - I apologize if I'm being difficult!

Reply:
I understand that, but the issue is, if I run `cargo check` with the following code

```
fn main() -> Result<()> {
    registry()
        .with(fmt::layer())
        .with(EnvFilter::from_default_env())
        .init();
    prompt_outdated_version();

    let cli = Cli::parse();
    match cli.commands {
        Commands::Exec(args) => exec(&args)?,
        Commands::Extract { path } => {
            extract_library(Some(path))?;
        }
    }
}
```

I receive the following error:

<a href="https://ibb.co/kHv9Y4L"><img src="https://i.ibb.co/p05RCfH/Screen-Shot-2022-06-03-at-2-44-03-PM.png" alt="Screen-Shot-2022-06-03-at-2-44-03-PM" border="0"></a>

Reply:
Sorry but the image isn't available.. you can upload directly to GitHub by pasting the image here.
Considering the snippet, I think I know what the issue is. You can just drop the `?` in both arms and it'd work because both `exec` and `extract_library` return `Result`, so no need to unwrap it and you can just return their return value.

Reply:
Added in your changes - since `extract_library` returns a `Result<String>` instead of `Result<()>`, I just return a normal `Ok(())`. I think that should cover everything!